game-core/src/main/java/games/strategy/triplea/TripleAPlayer.java

```diff
@@ -451,12 +450,6 @@ private void battle() {
       }
       final BattleListing battles = battleDel.getBattles();
       if (battles.isEmpty()) {
-        final IBattle battle = battleDel.getCurrentBattle();
-        if (battle != null) {
-          // this should never happen, but it happened once....
-          System.err.println("Current battle exists but is not on pending list:  " + battle.toString());
-          battleDel.fightCurrentBattle();
-        }
```

Comment:
This looks like it was an attempt to fix this issue in the past but now this should never occur anymore and this only fixed a small portion of the cases anyways.